### ISSUE TITLE
repoQuota Plugin is only working after the quota is exceeded but not checking before that

### ISSUE THREAD

**Issue by lsauer** [jfrog/artifactory-user-plugins#346]
As long as the Quota has not exceeded, one can still exceed the Quota by far, by uploading a very large file.
Only once the quota is already exceeded does the Quota mechanism work.
It would be great to define a upper tolerance limit, after which any upload fails. For instance, a repo still has 50MB left, the entire Quota is 1GB. Then tolerance is 5% of 1GB, which is 50MB. So If a user uploads now a file that exceeds 100MB then the upload should fail

**Comment by shashank-jfrog** [comment]
Hi,
Thank you for your query. Yes the ask in repoQuota plugin is not supported out of the box but we encourage to write a custom plugin on top of repoQuota using the beforeCreate event using [this signature](https://jfrog.com/help/r/jfrog-integrations-documentation/plugin-template-source) to solve your requirement. If you have any new questions or suggestions, please feel free to open a new issue.

Thanks
